### DISCUSSION TITLE
tutorials/federation-hotchocolate/09-contribute-fields

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#92] [course feedback]
# tutorials/federation-hotchocolate/09-contribute-fields

https://www.apollographql.com/tutorials/federation-hotchocolate/09-contribute-fields

**Comment by relshageaa** [comment]
where is randomRecipe declared, i am confused please

**Reply by lizhennessy** [reply]
Hi there @relshageaa! The `randomRecipe` field appears in the `recipes` subgraph we added to our graph in [Lesson 6: Adding a subgraph](https://www.apollographql.com/tutorials/federation-hotchocolate/06-add-subgraph). This brings the `recipes` subgraph schema into our supergraph so that we can query `recipes` at the same time that we query for `soundtracks`! You can see all of the available fields in your schema by opening up your graph is Studio and selecting the _Schema_ option from the menu on the left. Here you can also see which subgraph contributed the field! Please let us know if you have any other questions.

<img width="200" alt="access-schema" src="https://github.com/apollographql-education/content-feedback/assets/95302380/5a75c48c-a528-4547-bd60-8299a8091b37">
